.github/local-actions/branch-manager/lib/main.ts

```diff
@@ -0,0 +1,161 @@
+  const pullRequest = await loadAndValidatePullRequest(
+    {git, config},
+    pr,
+    new PullRequestValidationConfig(),
+  );
+  core.info('Validated PR information:');
+  core.info(JSON.stringify(pullRequest));
+  /** Whether any fatal validation failures were discovered. */
+  let hasFatalFailures = false;
+  /** The status information to be pushed as a status to the pull request. */
+  let statusInfo: CommmitStatus = await (async () => {
+    // Log validation failures and check for any fatal failures.
+    if (pullRequest.validationFailures.length !== 0) {
+      core.info(`Found ${pullRequest.validationFailures.length} failing validation(s)`);
+      for (const failure of pullRequest.validationFailures) {
+        hasFatalFailures = !failure.canBeForceIgnored || hasFatalFailures;
+        await core.group('Validation failures', async () => {
+          core.info(failure.message);
+        });
+      }
+    }
+
+    // With any fatal failure the check is not necessary to do.
+    if (hasFatalFailures) {
+      core.info('One of the validations was fatal, setting the status as pending for the pr');
+      return {
+        description: 'Waiting to check mergeability due to failing status(es)',
+        state: 'pending',
+      };
+    }
+
+    try {
+      git.run(['checkout', 'main']);
+      /**
+       * A merge strategy used to perform the merge check.
+       * Any concrete class implementing MergeStrategy is sufficient as all of our usage is
+       * defined in the abstract base class.
+       * */
+      const strategy = new AutosquashMergeStrategy(git);
+      await strategy.prepare(pullRequest);
+      await strategy.check(pullRequest);
+      core.info('Merge check passes, setting a passing status on the pr');
+      return {
+        description: `Merges cleanly to ${pullRequest.targetBranches.join(', ')}`,
+        state: 'success',
+      };
+    } catch (e) {
+      // As the merge strategy class will express the failures during checks, any thrown error is a
+      // failure for our merge check.
+      let description: string;
+      if (e instanceof MergeConflictsFatalError) {
+        core.info('Merge conflict found');
+        description = `Unable to merge into ${e.failedBranches.join(', ')}`;
+      } else {
+        core.info('Unknown error found when checking merge:');
+        core.error(e as Error);
+        description =
+          'Cannot cleanly merge to all target branches, please update changes or PR target';
+      }
+      return {
+        description,
+        state: 'failure',
+      };
+    }
+  })();
+
+  await git.github.repos.createCommitStatus({
+    ...repo,
+    state: statusInfo.state,
+    // Status descriptions are limited to 140 characters.
+    description: statusInfo.description.substring(0, 139),
+    sha: pullRequest.headSha,
+    context: 'mergeability',
+  });
+}
+
+/** The token for the angular robot to perform actions. */
+const token = await getAuthTokenFor(ANGULAR_ROBOT, true);
```

Comment:
Since the action only runs for a specific repo/PR, could it just use a repo-specific installation token, instead of introducing the org-wide token mechanism (we didn't need until this PR)

Reply:
The repo specific token will not have permission to set the status on other repos within the organization.

Reply:
Sorry, I might miss something. Couldn't you just request an installation token for the target repository? (provided via the action inputs)

Reply:
Yes, I don't know why I didn't think to do that.... will do.